fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java

```diff
@@ -516,6 +518,27 @@ public String getTableName() throws MetaNotFoundException {
         }
     }
 
+    // This method should be used by caller with other locks, like modifyJob()
+    public String getTableNameWithTimeout() throws UserException {
```

Comment:
1. tryGetTableName is better name?
2. This way is a workaround? The better way is to refactor the code logic to avoid deadlock rather than handling it so tricky.
We need to find out when the deadlock will happen, and fix it.

Reply:
The deadlock happens when a routine load commit txn comes, meanwhile, alter routine load cmd was executed:

commit txn thread：get db write lock firstly, then get job write lock
alter routine load job thread：get job write lock firstly, then got db read lock

Reply:
> alter routine load job thread：get job write lock firstly, then got db read lock

can you try to adjust the lock order, get db read lock first?

Reply:
@chen9t , @rickif  is the owner of routine load, he will help you to find a thorough solution.

Reply:
Remove read lock to fix this issue, please check it again. Thank you!